flink-runtime/src/main/java/org/apache/flink/runtime/leaderretrieval/ZooKeeperLeaderRetrievalService.java

```diff
@@ -170,7 +174,6 @@ public void nodeChanged() throws Exception {
 					notifyIfNewLeaderAddress(leaderAddress, leaderSessionID);
 				} catch (Exception e) {
 					leaderListener.handleError(new Exception("Could not handle node changed event.", e));
-					throw e;
```

Comment:
I assume the only implication of removing this throw is that curator won't log about the exception anymore, and it won't reset the interrupted flag.

I was wondering for a while if this has any implications, but I assume it is fine to remove, since the leader listener will handle the exception in Flink anyways.

Reply:
Yes, that is the reasoning. The `leaderListener.handleError` should handle all exceptions and usually terminate the process. But let me actually add `ExceptionUtils.checkInterrupted(e);` here again.